tools/webpack-jumpstart

```diff
@@ -44,18 +47,29 @@ if [ "${NODE_ENV-}" = "development" ]; then
     exit 1
 fi
 
-if ! git diff --quiet -- ':^test' ':^packit.yaml' ':^.github'; then
-    echo 'jumpstart: uncommitted local changes, skipping download' >&2
-    exit 1
```

Comment:
(Copied from #16158)

I really don't like dropping this check unconditionally. Now downloading the cache would claim to work even with local modifications, but then do that in vain and rebuild stuff anyway. To me as a developer this is too mushy and unpredictable -- either the cache applies to my tree or it doesn't. In the latter case, rebuilding webpacks is not really the end of the world.

I see two possible compromises:

 - Move this into the `else` branch  for the non-`--best` case
 - Drop the hardcoded "known safe" list (which I suppose is the one that annoys you?) and replace it with some `--ignore-local` or `--force` option. That makes stuff a wee bit harder for devs, but not unreasonably so.

Reply:
From this I understand that you're trying to say:

> After running jumpstart, without --best, typing `make` should never ever run webpack.  In the case that running webpack would be required, I'd rather have the jumpstart fail.

I think that's a reasonable property, and you're right that it's not currently the case.  I had intended to add a blanket check for changes in the local tree to prevent that unless `--best` was given (but your option 2), but I have another idea for a third option:

 - do everything, and in the case that any `dist/` directory would be missing, then fail.

That gives us a very very granular control over exactly which files the user is allowed to have modified (only the exact files which wouldn't result in rebuilding) and removes any arbitrary "this should be okay" type list-making.

It's also a pain in the butt to implement.  I'll let it simmer on the back burner...

Reply:
btw: in accordance with the above, I consider that maybe `--best` should be called `--partial` or so.  fwiw, it was originally called `--scan`.  I'm open to suggestions.

Reply:
I like `--partial`, I think it denotes the intention pretty well.

Reply:
I changed the name to `--partial`.

Reply:
I also made the check *stronger*.  It will reject any changes at all to the local tree, unless `--partial` is given.